### PR TITLE
feat: summer protocol client

### PR DESCRIPTION
**Pull request by halaprix**
Add summer-earn-protocol-subgraph package

This PR introduces the new summer-earn-protocol-subgraph package, which provides a typed GraphQL client for interacting with the Summer Earn Protocol subgraph. Key features and additions include:

- Typed GraphQL client generated using graphql-codegen
- Query for fetching user positions (UserPositions)
- TypeScript setup with build and dev scripts
- Integration with graphql-request for making GraphQL queries

Files added:
- src/index.ts: Main entry point for the package
- src/client.ts: Generated GraphQL client code
- queries.graphql: GraphQL query definitions
- package.json: Package configuration and dependencies
- tsconfig.json: TypeScript configuration
- README.md: Documentation for usage and development

Next steps:
- Set up automated testing
- Add more queries as needed

## Summary by CodeRabbit

- **New Features**
	- Enhanced workflow configurations to securely access sensitive environment variables during build processes.
	- Added a script to automate the restoration of Git submodules, improving project setup and maintenance.
	- Introduced a new configuration file for GraphQL code generation, streamlining TypeScript integration.
	- Implemented a client for querying user positions from a subgraph, enabling robust data retrieval across blockchain networks.

- **Chores**
	- Established a consistent approach to managing environment variables across various GitHub Actions workflows for improved security and flexibility.

### CODE REVIEW

File: packages/summer-earn-protocol-subgraph/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        throw new Error(`ChainId ${chainId} not supported
+        Supported chains: ${supportedChains}`)
+    }
+    const subgraph = chainIdSubgraphMap[chainId]
+    if (!subgraph) {
+        throw new Error(`No subgraph for chainId ${chainId}`)
+    }
+    const url = `${baseUrl}/${subgraph}`
+    const client = new GraphQLClient(url);
+    return getSdk(client);
```
Comment (robercano) [inline packages/summer-earn-protocol-subgraph/src/index.ts:31]: Is this SDK related somehow to our SDK?

File: packages/summer-events-subgraph/README.md
Comment (robercano) [inline packages/summer-events-subgraph/README.md:1]: 👏 Amazing!

**Review by robercano — APPROVED** [review]
Looks good to me, just a minor question

File: packages/summer-earn-protocol-subgraph/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "@graphql-codegen/cli": "^5.0.2",
+    "@graphql-codegen/typescript": "^4.0.9",
+    "@graphql-codegen/typescript-graphql-request": "^6.2.0",
+    "@graphql-codegen/typescript-operations": "^4.2.0",
+    "@summerfi/abstractions": "workspace:*",
+    "@summerfi/serverless-shared": "workspace:*"
+  },
+  "dependencies": {
+    "graphql-request": "^6.1.0",
+    "tsx": "^4.16.2",
```
Comment (piotrwitek) [inline packages/summer-earn-protocol-subgraph/package.json:29]: do we need tsx as dependency?

File: packages/summer-earn-protocol-subgraph/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@graphql-codegen/cli": "^5.0.2",
+    "@graphql-codegen/typescript": "^4.0.9",
+    "@graphql-codegen/typescript-graphql-request": "^6.2.0",
+    "@graphql-codegen/typescript-operations": "^4.2.0",
+    "@summerfi/abstractions": "workspace:*",
+    "@summerfi/serverless-shared": "workspace:*"
```
Comment (piotrwitek) [inline packages/summer-earn-protocol-subgraph/package.json:25]: this should be dep, as it's used in src

File: packages/summer-earn-protocol-subgraph/src/client.ts [hunk 4]
```diff
@@ -0,0 +1,4109 @@
+// @ts-nocheck
+// This file was automatically generated and should not be edited.
+import { GraphQLClient, RequestOptions } from 'graphql-request';
+import gql from 'graphql-tag';
```
Comment (piotrwitek) [inline packages/summer-earn-protocol-subgraph/src/client.ts:4]: that's missing dep in the pkg.json

File: packages/summer-earn-protocol-subgraph/package.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "dependencies": {
+    "graphql-request": "^6.1.0",
+    "tsx": "^4.16.2",
+    "typescript": "^5.4.5"
```
Comment (piotrwitek) [inline packages/summer-earn-protocol-subgraph/package.json:30]: typescript should be devDep

**Review by piotrwitek — COMMENTED** [review]
Looks super duper cool :) Left few comments with some stuff that I noticed, also just wanted to remind about missing eslint config. Cheers 🫶

File: packages/summer-earn-protocol-subgraph/src/client.ts (hunk 4 above)
Comment (halaprix) [reply packages/summer-earn-protocol-subgraph/src/client.ts:4]: not sure if it's required ?
the client is autogenerated and the dependency is the dependency of library that generates the client.

btw not sure why the autogenerated client is not importing `gql` directly from `graphql-request`

File: packages/summer-earn-protocol-subgraph/src/index.ts (hunk 1 above)
Comment (halaprix) [reply packages/summer-earn-protocol-subgraph/src/index.ts:31]: no. it's a method in the autogenerated graphql cleint :)

File: packages/summer-earn-protocol-subgraph/src/client.ts (hunk 4 above)
Comment (piotrwitek) [reply packages/summer-earn-protocol-subgraph/src/client.ts:6]: if autogenerated then should be good

File: packages/summer-earn-protocol-subgraph/package.json (hunk 2 above)
Comment (piotrwitek) [reply packages/summer-earn-protocol-subgraph/package.json]: cool :D

**Review by piotrwitek — APPROVED** [review]
LGTM

**Pull request merged**